### PR TITLE
Improve performance of File block processor

### PR DESCRIPTION
**Pull request by fluiddot**
This PR follows the same approach taken in https://github.com/wordpress-mobile/WordPress-iOS/pull/22896 to improve the performance of Gutenberg processors. The overall improvement will be less significant in this case (that's why the PR is marked as `Enhancement` instead of `Bug`) mainly because the File block is used by fewer users than Image/Gallery blocks. Nevertheless, it will speed up the post-saving process.

**To test:**

Follow the testing instructions to cover the functionality related to the processors for the block:

* File Block:
  * [Close the post with an ongoing file upload](https://github.com/wordpress-mobile/test-cases/blob/trunk/test-cases/gutenberg/file.md#tc005)
  * [Close/Re-open post with an ongoing file upload](https://github.com/wordpress-mobile/test-cases/blob/trunk/test-cases/gutenberg/file.md#tc012)

## Regression Notes
1. Potential unintended areas of impact
The block processors are only used when saving a post. No other area should be impacted.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested saving a post with File blocks.

3. What automated tests I added (or what prevented me from doing so)
I added a new unit test to cover the content parsing of File blocks.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/ViewRelated/Gutenberg/Processors/GutenbergFileUploadProcessor.swift [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import Foundation
-import Aztec
```
Comment (fluiddot) [inline WordPress/Classes/ViewRelated/Gutenberg/Processors/GutenbergFileUploadProcessor.swift:2]: Aztec was imported in this file to use `HTMLProcessor`. Now the HTML processing is performed by using `SwiftSoup`.

**Comment by irfano** [comment]
I'm moving this to the next milestone since this is not a blocker, and the code freeze will be completed today.

**Comment by momo-ozawa** [comment]
👋 Hey @fluiddot, I'm bumping this PR to 24.8 since it's code freeze day. If this PR needs to target 24.7, please target the release branch once it's been cut. Thanks!

**Comment by dvdchr** [comment]
Hi @fluiddot 👋 , I'm bumping this PR's milestone to `24.9` since I'm starting code freeze. Feel free to re-target this to the release branch if this is a blocker or intended for `24.8`.

**Comment by salimbraksa** [comment]
Hey @fluiddot, I'm bumping the milestone for this PR to `25.0` as I'm starting the code freeze. If this PR needs to go into `24.9`, please go ahead and re-target it to the release branch.

**Review by twstokes — APPROVED** [review]
LGTM @fluiddot! Thanks for adding the tests. 🚀

**Pull request merged**